src/skmatter/neighbors/_sparsekde.py

```diff
@@ -0,0 +1,601 @@
+import warnings
+from typing import Union
+
+import numpy as np
+from scipy.special import logsumexp as LSE
+from sklearn.base import BaseEstimator
+from sklearn.utils.validation import check_is_fitted, check_random_state
+from tqdm import tqdm
+
+from ..metrics._pairwise import (
+    pairwise_euclidean_distances,
+    pairwise_mahalanobis_distances,
+)
+from ..utils._sparsekde import effdim, oas
+
+
+DIST_METRICS = {
+    "periodic_euclidean": pairwise_euclidean_distances,
+}
+
+
+class SparseKDE(BaseEstimator):
+    """A sparse implementation of the Kernel Density Estimation.
+    This class is used to build a sparse kernel density estimator.
+    It takes a set of descriptors and a set of weights as input,
+    and fit the KDE model on the sampled point (e.g. the grid point
+    selected by FPS). First, the probability density is estimated for
+    each sampled point. Then, quick shift clustering is applied to the
+    grid points. Finally, a kernel density estimator is built based on
+    the clustering results.
+
+    Parameters
+    ----------
+    descriptors: np.ndarray
+        Descriptors of the system where you want to build a sparse KDE.
+        It should be an array of shape `(n_descriptors, n_features)`.
+    weights: np.ndarray, default=None
+        Weights of the descriptors.
+        If None, all weights are set to `1/n_descriptors`.
+    kernel : {'gaussian'}, default='gaussian'
+        The kernel to use. Currentlty only one.
+    metric : str, default='periodic_euclidean'
+        The metric to use. Currently only one.
+    metric_params : dict, default=None
+        Additional parameters to be passed to the use of
+        metric.  i.e. the cell dimension for `periodic_euclidean`
+        {'cell': [2, 2]}
```

Comment:
I think here we should also give the function instead of a string. The params is a very good idea!